assert/assertion_order.go

```diff
@@ -0,0 +1,81 @@
+package assert
+
+import (
+	"fmt"
+	"reflect"
+)
+
+// isOrdered checks that collection contains orderable elements.
+func isOrdered(t TestingT, object interface{}, allowedComparesResults []CompareType, failMessage string, msgAndArgs ...interface{}) bool {
+	objKind := reflect.TypeOf(object).Kind()
+	if objKind != reflect.Slice && objKind != reflect.Array {
+		return false
+	}
+
+	objValue := reflect.ValueOf(object)
+	objLen := objValue.Len()
+
+	if objLen <= 1 {
+		return true
+	}
+
+	value := objValue.Index(0)
+	valueInterface := value.Interface()
+	firstValueKind := value.Kind()
```

Comment:
Given that in lines 18-20 we've guaranteed that the array will be of at least length 2 by the time we get to this point, would it not be clearer if we just moved this code inside the for loop and referenced the previous index?

e.g.
```
for i:= 1; i < objLen; i++ {
    prevValue := objValue.Index(i-1)
    value := objValue.Index(i)
    ...
}
...
```

Reply:
You could also inline the assignments that way in cases where it would improve clarity.